src/Pos/Client/Txp/Util.hs

```diff
@@ -206,60 +260,184 @@ prepareInpsOuts utxo addrs outputs = do
+    feePolicy <- bvdTxFeePolicy <$> lift gsAdoptedBVData
+    case feePolicy of
+        TxFeePolicyUnknown w _ -> throwTxError $
+            sformat ("Unknown fee policy, tag: "%build) w
+        TxFeePolicyTxSizeLinear linearPolicy ->
+            stabilizeTxFee linearPolicy utxo outputs
+
+-- Read-only function.
+computeTxFee
+    :: TxDistrMode ctx m
+    => Utxo
+    -> TxOutputs
+    -> ExceptT TxError m TxFee
+computeTxFee utxo outputs = do
+    feePolicy <- bvdTxFeePolicy <$> gsAdoptedBVData
+    case feePolicy of
+        TxFeePolicyUnknown w _ -> throwTxError $
+            sformat ("Unknown fee policy, tag: "%build) w
+        TxFeePolicyTxSizeLinear linearPolicy -> do
+            txAux <- createFakeTxFromRawTx <$>
+                     stabilizeTxFee linearPolicy utxo outputs
+            txToLinearFee linearPolicy txAux
+
+-- | Search such spendings that transaction's fee would be stable.
+stabilizeTxFee
+    :: forall m. MonadError TxError m
+    => TxSizeLinear
+    -> Utxo
+    -> TxOutputs
+    -> m TxRaw
+stabilizeTxFee linearPolicy utxo outputs =
+    stabilizeTxFeeDo 5 (TxFee $ mkCoin 0)
+  where
+    stabilizeTxFeeDo :: Int -> TxFee -> m TxRaw
+    stabilizeTxFeeDo 0 _ = throwTxError "Couldn't stabilize tx fee after 5 attempts"
+    stabilizeTxFeeDo attempt expectedFee = do
+        txRaw <- prepareTxRaw utxo outputs expectedFee
+        txFee <- txToLinearFee linearPolicy $
+                 createFakeTxFromRawTx txRaw
+        if expectedFee == txFee
+            then pure txRaw
+            else stabilizeTxFeeDo (attempt - 1) txFee
+
+txToLinearFee
+    :: MonadError TxError m
+    => TxSizeLinear -> TxAux -> m TxFee
+txToLinearFee linearPolicy =
+    either throwError pure .
+    bimap invalidFee TxFee .
+    integerToCoin .
+    ceiling .
+    calculateTxSizeLinear linearPolicy .
+    biSize @TxAux
+  where
+    invalidFee reason = TxError ("Invalid fee: " <> reason)
+
+createFakeTxFromRawTx :: TxRaw -> TxAux
+createFakeTxFromRawTx TxRaw{..} =
+    let fakeAddr = txOutAddress . toaOut . NE.head $ trOutputs
+        fakeOutMB
+            | trRemaining == mkCoin 0 = Nothing
+            | otherwise = Just $ TxOutAux (TxOut fakeAddr trRemaining) []
+        txOutsWithRem = maybe trOutputs (\remTx -> remTx :| toList trOutputs) fakeOutMB
+
+        -- We create fake signers instead of safe signers,
+        -- because safe signer requires passphrase
+        -- but we don't want to reveal our passphrase to compute fee.
+        -- Fee depends on size of tx in bytes, sign of a tx has the fixed size
+        -- so we can use arbitrary signer.
+        srcAddrs = NE.map fst trInputs
+        (_, fakeSK) = deterministicKeyGen "patakbardaqskovoroda228pva1488kk"
+        hdwSigners = NE.zip (NE.repeat $ fakeSigner fakeSK) srcAddrs
+    in makeMPubKeyTxAddrs hdwSigners trInputs txOutsWithRem
+
+    -- buildDistribution :: TxRaw -> Text
+    -- buildDistribution TxRaw{..} =
+    --     let entries =
```

Comment:
This function was used only for logging in the old code. Now the code, identical to old one, is run in environment without `WithLogger`. Is it better to add `WithLogger` to `TxDistrMode` or can we simply abandon this log line?